### PR TITLE
[#434] Empty Collection State

### LINKED ISSUE #434: Empty collection state for tvOS
Show text indicating when a collection is empty (no products). Also allow the Focus Engine to focus onto that collection even if its empty.

### PR DESCRIPTION
**Pull request by EishanVijay**
Added label to indicate when a collection was empty (no products). Also allowed the Focus Engine to focus on an empty collection by providing it an empty product cell to focus to.

Resolves #434 

@davidmuzi @bgulanowski

### CODE REVIEW

File: Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/CollectionViewContainerCell.swift [hunk 1]
```diff
@@ -32,14 +32,23 @@ class CollectionViewContainerCell: UICollectionViewCell {
...
+        super.prepareForReuse()
+        self.emptyCollectionViewLabel.isHidden = false
+    }
+    
     func configure(viewModel: ProductsViewModel, title: String) {
         self.products = viewModel
         self.collectionView.dataSource = self.products
         self.titleLabel.text = title
         self.products.getProducts {
+            if self.products.hasProducts {
```
Comment (davidmuzi) [inline Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/CollectionViewContainerCell.swift]: self.emptyCollectionViewLabel.isHidden = self.products.hasProducts

File: Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift [hunk 2]
```diff
@@ -58,21 +62,26 @@ extension ProductsViewModel: UICollectionViewDataSource {
     }
     
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return self.products.count
+        return (self.products.count > 0) ? self.products.count : 1
```
Comment (davidmuzi) [inline Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift]: return hasProducts ? self.products.count : 1

File: Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift [hunk 3]
```diff
@@ -58,21 +62,26 @@ extension ProductsViewModel: UICollectionViewDataSource {
...
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ProductCollectionViewCell.reuseIdentifier, for: indexPath) as? ProductCollectionViewCell
-        let product = self.products[indexPath.row] as BUYProduct
-        if product.imagesArray().count > 0 {
-            cell?.productImage.load(product.imagesArray().first, animateChange: true, completion: nil)
+        if products.count > 0 {
```
Comment (davidmuzi) [inline Mobile Buy SDK Sample Apps/Mobile Buy SDK tvOS Sample/Mobile Buy SDK tvOS Sample/ProductsViewModel.swift]: hasProducts

**Comment by EishanVijay** [comment]
Whoops I made the `hadProducts` variable later on and forgot to refactor prev code

**Pull request closed without merging**